### DISCUSSION TITLE
Is there a log stream function?

### DISCUSSION THREAD

**Discussion by traveltek-tmalek** [allinurl/goaccess#2252] [Q&A]
We have Ubuntu servers coming in and out of service within AWS from an ASG. So we'd need this installed on all servers built into the image. How do we see all the data from all servers (log streams) as we couldn't feasibly connect to each server to get this data.

Is this possible? I couldn't see it mentioned on the website. Thanks.

**Comment by allinurl** [comment]
Let me see if I follow your question, are you trying to get all your access logs from multiple servers read by a single instance of goaccess? or goaccess would be installed on each server and will individually read the access log individually? Thanks

**Reply by traveltek-tmalek** [reply]
Thanks for your response. So the servers are all ephemeral. Servers will spin up and spin down depending on the Scaling group schedule and various scaling policies.

I'm just trying to think if Go Access has a feature which allows us to view all our log streams/data in one place.

I don't know if you're familiar with AWS CloudWatch, but every server that ever existed creates it's own Log stream in the CloudWatch service in AWS. In the case of Go Access, I'm wondering if we "bake" in the Go Access install into every server that comes into service, and the metrics/data automatically get sent to a central place.

Sort of like having a web portal where we can log in and all the servers that are pushing data can be viewed here. Rather than individually logging into each server and looking at the data in a per server basis.

**Reply by allinurl** [reply]
GoAccess doesn't handle the transmission of the logs. It only consumes those logs. You will need to set up something in your infrastructure to have them in one place so goaccess can parse them. You could set up a centralize log server. You can have all your web servers to log traffic to a remote server. [**This**](http://rafaelsteil.com/apache-remote-logging-with-rsyslog/) is probably a bit old, but it might be helpful to get started.

All entries coming from your web servers will be put under the file you specified in:

```
$template apacheAccess,"/var/log/apache2/access.log"
```

Then you can have goaccess parse that specific log on the remote server.

**Reply by traveltek-tmalek** [reply]
Thank you :)
